cmd/dump/cmd-to-md.go

```diff
@@ -0,0 +1,247 @@
+// Copyright (c) 2017-present SIGHUP s.r.l All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package dump
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+// Code blow is adapted from:
+// https://github.com/spf13/cobra/blob/main/doc/md_docs.go
+// https://github.com/spf13/cobra/blob/main/doc/util.go
+
+// Copyright 2013-2023 The Cobra Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// Copyright 2013-2023 The Cobra Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const markdownExtension = ".md"
+
+func printOptions(buf *bytes.Buffer, cmd *cobra.Command, _ string) error {
+	flags := cmd.NonInheritedFlags()
+	flags.SetOutput(buf)
+
+	if flags.HasAvailableFlags() {
+		if _, err := buf.WriteString("## Options\n\n```bash\n"); err != nil {
+			return fmt.Errorf("error while writing to buffer: %w", err)
+		}
+
+		flags.PrintDefaults()
+
+		if _, err := buf.WriteString("```\n\n"); err != nil {
+			return fmt.Errorf("error while writing to buffer: %w", err)
+		}
+	}
+
+	parentFlags := cmd.InheritedFlags()
+	parentFlags.SetOutput(buf)
+
+	if parentFlags.HasAvailableFlags() {
+		if _, err := buf.WriteString("## Options inherited from parent commands\n\n```bash\n"); err != nil {
+			return fmt.Errorf("error while writing to buffer: %w", err)
+		}
+
+		parentFlags.PrintDefaults()
+
+		if _, err := buf.WriteString("```\n\n"); err != nil {
+			return fmt.Errorf("error while writing to buffer: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// GenMarkdown creates markdown output.
+func GenMarkdown(cmd *cobra.Command, w io.Writer) error {
```

Comment:
Do we really need to pass an `io.Writer` here? Shouldn't we just output a string and then leave to the caller the decision of what to do with it?

Reply:
Removed function altogether. We were not using it